docs/QuickStart.md

```diff
@@ -0,0 +1,204 @@
+    
+    spider::Data<HdfsFile> output = spider::Data<HdfsFile>::Builder()
+        .cleanup([](HdfaFile const& file) { delete_hdfs_file(file); })
+        .build(HdfsFile { output_path });
+    
+    run_map(input_path, output_path);
+    
+    // Now that map finishes, the file is persisted on Hdfs as output of job.
+    output.mark_persist();
+    return output;
+}
+
+auto main(int argc, char** argv) -> int {
+    spider::Data<HdfsFile> input = spider::Data<HdfsFile>::Builder()
+        .mark_persist(true)
+        .build(HdfsFile { "/path/to/input" });
+    spider::Future<spider::Data<HdfsFile>> future = spider::run(
+        spider::bind(map, filter),
+        input);
+    std::string const output_path = future.get().get().url;
+    std::cout << "Result is stored in " << output_path << std::endl;
+}
+```
+
+## Data as Key-Value Store
+`Data` can also be used a a key-value store. User can specify a key when creating the data, and the data can be accessed later by its key. Notice that a task can only access the `Data` created by itself or passed to it. Client can access any data with the key.
+Using the key value store, we can solve the dynamic task result problem.
+
+```c++
+auto gcd(spider::Context& context, int x, int y, std::string key)
+    -> std::tuple<int, int, std::string> {
+    if (x == y) {
+        spider::Data<int>.Builder()
+            .set_key(key)
+            .build(x);
+        return { x, y, key };
+    }
+    if (x > y) {
+        context.add_child(gcd);
+        return { x % y, y, key };
+    }
+    context.add_child(gcd);
+    return { x, y % x, key };
+}
+
+auto main(int argc, char** argv) -> int {
+    std::string const key = "random_key";
+    driver.run(gcd, 48, 18, key);
+    while (!driver.get_data_by_key(key)) {
+        int value = driver.get_data_by_key(key).get();
+        std::cout << "gcd of " << x << " and " << y << " is " << value << std::endl;
+    }
+}
+```
+
+## Straggler Mitigation
+`Driver::register_task` can take a second argument for timeout milliseconds. If a task executes for longer than the specified timeout, Spider spawns another task instance running the same function. The task that finishes first wins. Other running task instances are cancelled, and associated data is cleaned up.
+
+The new task has a different task id, and it is the responsibility of the user to avoid any data race and deduplicate the output if necessary.
+
+## Note on Worker Setup
+The setup section said that we can start a worker by running `spider start --worker --db <db_url>`. This is oversimplified. The worker has to know the function it will run.
+
+When user compiles the client code, an executable and a library are generated. The executable executes the client code as expected. The library contains all the functions registered by user. Worker needs to run with a copy of this library. The actual commands to start a worker is `spider start --worker --db <db_url> --libs [client_libraries]`.
```

Comment:
This makes sense, so in the guide above, wouldn't it be easier to just say that we need to create a task library with all the tasks the user wants to run? Then we can reference those tasks in the client code. Right now, the guide says we need to register the tasks by calling `spider::register_task` which sounds pretty opaque to me.

Reply:
User is responsible to create a task library and the client executable. Actually a good practice for user is to put the `spider::register_task` with the tasks' declaration instead of main, so maybe I should task registration into "Create a task section?